$OpenBSD: patch-libs_filesystem_src_operations_cpp,v 1.5 2021/02/01 12:00:51 rsadowski Exp $

On OpenBSD 4.4 and newer, use statvfs to obtain filesystem space info.

Index: libs/filesystem/src/operations.cpp
--- libs/filesystem/src/operations.cpp.orig
+++ libs/filesystem/src/operations.cpp
@@ -68,6 +68,7 @@
 #endif
 
 #include <boost/filesystem/config.hpp>
+#include <boost/predef/os/bsd/open.h>
 #include <boost/filesystem/operations.hpp>
 #include <boost/filesystem/file_status.hpp>
 #include <boost/filesystem/exception.hpp>
@@ -97,8 +98,7 @@
 
 #   include <sys/types.h>
 #   include <sys/stat.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__) \
- && !defined(__VXWORKS__)
+#   if !defined(__APPLE__) && (!defined(__OpenBSD__) || BOOST_OS_BSD_OPEN >= BOOST_VERSION_NUMBER(4, 4, 0)) && !defined(__ANDROID__) && !defined(__VXWORKS__)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
